documentation/source/KD.md

```diff
@@ -0,0 +1,119 @@
+Equivalently, for knowledge distillation we use the `KDTrainer` class which inherits from `Trainer`.
+
+If for regular training with `Trainer`, the general flow is:
+
+```python
+...
+trainer = Trainer("my_experiment")
+train_dataloader = ...
+valid_dataloader = ...
+model = ...
+
+train_params = {...}
+trainer.train(model=model, training_params=train_params, train_loader=train_dataloader, valid_loader=valid_dataloader)
+```
+
+
+Then for training with knowledge distillation, the general flow is:
+```python
+
+from super_gradients.training.kd_trainer import KDTrainer
+...
+kd_trainer = KDTrainer("my_experiment")
+train_dataloader = ...
+valid_dataloader = ...
+
+student_model = ...
+teacher_model = ...
+
+train_params = {...}
+kd_trainer.train(student=student_model, teacher=teacher_model, training_params=train_params, train_loader=train_dataloader, valid_loader=valid_dataloader)
+```
+
+Check out our [knowledge distillation tutorial notebook](https://bit.ly/3BLA5oR) to see a practical example in action.
+
+
+## Knowledge Distillation Training in SG: Key Components
+
+
+### [KDModule](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/training/models/kd_modules/kd_module.py)
+
+The most apparent difference in the flow of training using knowledge distillation is the fact that it requires two networks: the "teacher" and the "student".
+The relation between the two is also configurable - for example, we may decide that the teacher model should preprocess the inputs differently.
+For that matter, SG introduces a new `torch.nn.Module` that wraps both the student and the teacher models: `KDModule`.
+Uppon calling `KDTrainer.train()` the teacher and student models are passed, along the `kd_arch_params` to initialize a `KDModule` instance.
+Passing a `KDModule` instance explicitly to `KDTrainer.train()` through the `model` argument instead of student and teacher models is also possible, which gives the users the option to customize KD to their needs.
+
+### [KDOutput](https://github.com/Deci-AI/super-gradients/blob/12a4e53a96e8608409100b5ef83971157518434b/src/super_gradients/training/models/kd_modules/kd_module.py#L7)
+
+`KDOutput` defines the structure of the output of `KDModule`, and has two self-explanatory attributes: student_output, teacher_output.
+`KDTrainer` uses these attributes behind the scenes to perform the usual operations of normal training such as metrics calculations.
+This means that when customizing KD, its important for the custom `KDModule` to stick to this output format.
+
+### KD Losses
+
+Currently, [KDLogitsLoss](https://github.com/Deci-AI/super-gradients/blob/12a4e53a96e8608409100b5ef83971157518434b/src/super_gradients/training/losses/kd_losses.py#L15) is the only in SGs KD losses bank, but more are to come.
```

Comment:
typo: "the only in" => "the only one in"

Reply:
fixed